src/functions.jl

```diff
@@ -146,6 +146,78 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     return NamedDimsArray{newL}(data)
 end
 
+################################################
+# cat, vcat, hcat
+
+function Base.cat(a::NamedDimsArray{L}; dims) where L
+    newL = expand_dimnames(L, dims)
+    numerical_dims = dim(newL, dims)
+    data = Base.cat(parent(a); dims=numerical_dims)
+    newL = unify_names_longest(newL, dimnames(data))
+    return NamedDimsArray{newL}(data)
+end
+
+function Base.cat(a::NamedDimsArray{L}, b::AbstractArray; dims) where L
+    newL = expand_dimnames(L, dims)
+    numerical_dims = dim(newL, dims)
+    data = Base.cat(parent(a), b; dims=numerical_dims)
+    newL = unify_names_longest(newL, dimnames(data)) # when dims=3 for two 2d arrays
+    return NamedDimsArray{newL}(data)
+end
+
+function Base.cat(a::AbstractArray, b::NamedDimsArray{L}; dims) where L
+    newL = expand_dimnames(L, dims)
+    numerical_dims = dim(newL, dims)
+    data = Base.cat(a, parent(b); dims=numerical_dims)
+    newL = unify_names_longest(newL, dimnames(data)) # when dims=3 for two 2d arrays
+    return NamedDimsArray{newL}(data)
+end
+
+function Base.cat(a::NamedDimsArray{La}, b::NamedDimsArray{Lb}; dims) where {La, Lb}
+    newL = unify_names_shortest(La, Lb)
+    newL = expand_dimnames(newL, dims)
+    numerical_dims = dim(newL, dims)
+    data = Base.cat(parent(a), parent(b); dims=numerical_dims)
+    newL = unify_names_longest(newL, dimnames(data)) # when dims=3 for two 2d arrays
+    return NamedDimsArray{newL}(data)
+end
+
+# to dispatch on the first or the second argument being the NDA
+for (T, S) in [
+    (:NamedDimsArray, :AbstractArray),
+    (:AbstractArray, :NamedDimsArray),
+    (:NamedDimsArray, :NamedDimsArray)
+    ]
+
+    @eval function Base.cat(a::$T, b::$S, c::AbstractArray...; dims)
+        return Base.cat(a, Base.cat(b, Base.cat(c...; dims=dims); dims=dims); dims=dims)
+    end
+
+end
+
+# Base.hcat and Base.vcat specialise on this Union
+const AbsVecOrMat = Union{AbstractVector, AbstractMatrix}
+for (T, S) in [
+    (:NamedDimsArray, :AbsVecOrMat),
+    (:NamedDimsArray, :AbstractArray),
+    (:AbsVecOrMat, :NamedDimsArray),
+    (:AbstractArray, :NamedDimsArray),
+    (:NamedDimsArray, :NamedDimsArray)
+    ]
+
+    for (fun, d) in zip((:vcat, :hcat), (1, 2))
+
+        @eval function Base.$fun(a::$T, b::$S)
+            return Base.cat(a, b, dims=$d)
```

Comment:
per the style-guide:
https://github.com/invenia/BlueStyle#keyword-arguments

```suggestion
            return Base.cat(a, b; dims=$d)
```